src/pages/collections/[collectionAddress]/activity.tsx

```diff
@@ -0,0 +1,321 @@
+import {
+  CollectionPageProps,
+  getCollectionPageServerSideProps,
+} from '@/views/collections/collections.utils';
+
+import CollectionLayout from 'src/views/collections/CollectionLayout';
+import { InView } from 'react-intersection-observer';
+
+import { GetServerSideProps } from 'next';
+
+import { uniq } from 'ramda';
+import React, { FC, ReactNode, useState } from 'react';
+import { NftActivity, useCollectionActivitiesQuery } from 'src/graphql/indexerTypes';
+import TopLevelFilterButton from 'src/components/TopLevelFilterButton';
+import clsx from 'clsx';
+import NoProfileItems, { NoProfileVariant } from '../../../components/NoProfileItems';
+import { ApolloQueryResult, OperationVariables } from '@apollo/client';
+import { None } from '../../../components/OfferForm';
+import { TailSpin } from 'react-loader-spinner';
+import { format } from 'timeago.js';
+import Link from 'next/link';
+import { DollarSign } from 'react-feather';
+import { DisplaySOL } from '../../../components/CurrencyHelpers';
+import { Avatar } from '../../../components/Avatar';
+import { HandIcon, TagIcon } from '@heroicons/react/outline';
+import { shortenAddress } from '../../../modules/utils/string';
+
+export const getServerSideProps: GetServerSideProps<CollectionPageProps> =
+  getCollectionPageServerSideProps;
+
+export const LoadingActivityCard = () => {
+  return <article className="mb-4 h-16 rounded bg-gray-800" />;
+};
+
+export const ActivityCard = ({
+  activity: a,
+  refetch,
+  loading = false,
+}: {
+  activity: NftActivity;
+  refetch:
+    | ((variables?: Partial<OperationVariables> | undefined) => Promise<ApolloQueryResult<None>>)
+    | (() => void);
+  loading: boolean;
+}) => {
+  if (loading) return <LoadingActivityCard />;
+  const multipleWallets = a.wallets.length > 1;
+  return (
+    <article key={a.id} className="mb-4 grid grid-cols-8 rounded border border-gray-700 p-4">
+      <Link href={`/nfts/${a.nft?.address}`} passHref>
+        <a className="col-span-2 flex items-center gap-4">
+          <img className="h-12 w-12 rounded-lg object-cover" src={a.nft?.image} alt="nft" />
+          <div className="font-medium">{a.nft?.name}</div>
+        </a>
+      </Link>
+      <div className="flex items-center">
+        {a.activityType === 'purchase' && (
+          <DollarSign className="mr-2 h-5 w-5 self-center text-gray-300" />
+        )}
+        <div>{a.activityType === 'purchase' && 'Sold'}</div>
+        {a.activityType === 'offer' && (
+          <HandIcon className="mr-2 h-5 w-5 self-center text-gray-300" />
+        )}
+        <div>{a.activityType === 'offer' && 'Offer'}</div>
+        {a.activityType === 'listing' && (
+          <TagIcon className="mr-2 h-5 w-5 self-center text-gray-300" />
+        )}
+        <div>{a.activityType === 'listing' && 'Listing'}</div>
+      </div>
```

Comment:
No need to use ramda but look to assign items to variables instead of a bunch of conditional renders.

https://github.com/holaplex/marketplace/blob/dev/src/layouts/Nft/NftLayout.tsx#L432